scripts/mbedtls_dev/bignum_common.py

```diff
@@ -150,6 +152,53 @@ def get_value_pairs(cls) -> Iterator[Tuple[str, str]]:
 # END MERGE SLOT 6
 
 # BEGIN MERGE SLOT 7
+    def arch_specific(self):
+        """ Placeholder for architecture specific patching
+        """
+        pass
+
+def multiarch(class_override):
```

Comment:
That is a good point, we could have two constructors, one of them taking an architecture parameter and the other one not.

Reply:
Python doesn't support multiple constructors (there's only `__init__`), but we can have an optional named argument on the one constructor.

Reply:
The core issue is that we effect we need:

* To override specific class methods
* To add extra class member properties if that setting is enabled
* Adjust the intiatiation arguments to include the bits_in_limb parameter, without redesigning the whole inheritance model.

The problem is that the tool is dynamically generating classes throuhg inheritance and metaclasses.

~~~~~~
    @classmethod
    def generate_function_tests(cls) -> Iterator[test_case.TestCase]:
        """Override for additional scalar input."""
        for a_value, b_value in cls.get_value_pairs():
            for s_value in cls.input_scalars:
                cur_op = cls(a_value, b_value, s_value)
                yield cur_op.create_test_case()
~~~~~~

Because you canot subclass from multiple metaclasses, and the generation code above is using (input_a, input_b) pairs and create_test_case to dynamically instantiate the classes, **we are effectively injecting the bits_per_limb parameter**.  

 The Class thiis feature is adding functionality from is bringing in features from OperationCommon and BaseTarget

~~~~~~~
BignumCoreOperation(bignum_common.OperationCommon, BignumCoreTarget, metaclass=ABCMeta)
~~~~~~~

Decorating the resul of it, in order to add the needed functionality is the only solution that meets the original ask of keeping it simple and elegant. 

Building the functionality in `OperationCommon` would be ideal but would require restructuring of the tool, and `BignumCommonOperation` should also be required, as to where the `BignumCoreOperation, BignumModRawOperation`  would inherit from.